lib/Prometheus/Tiny.pm

```diff
@@ -43,13 +44,21 @@ sub _format_labels {
 
 sub set {
   my ($self, $name, $value, $labels, $timestamp) = @_;
+  unless (looks_like_number $value) {
+    carp "setting '$name' to non-numeric value, using 0 instead";
+    $value = 0;
+  }
   my $f_label = $self->_format_labels($labels);
   $self->{metrics}{$name}{$f_label} = [ $value, $timestamp ];
   return;
 }
 
 sub add {
   my ($self, $name, $value, $labels) = @_;
+  unless (looks_like_number $value) {
+    carp "adjusting '$name' by non-numeric value, using 0 instead";
```

Comment:
I would change this to `adding 0` instead of `using 0`, which I think is clearer. ("using 0" could mean you're now using 0 as the new value, even if it was 10 and you're adding undef; in this case, the metric will be 10, not 0.)